src/errors.rs

```diff
@@ -14,22 +12,29 @@ pub type BcryptResult<T> = Result<T, BcryptError>;
 /// All the errors we can encounter while hashing/verifying
 /// passwords
 pub enum BcryptError {
-    #[cfg(feature = "std")]
-    Io(io::Error),
+    /// Raised when the cost value is outside of the allowed 4-31 range.
+    ///
+    /// Cost is provided as an argument to hashing functions, and extracted from the hash in
+    /// verification functions.
```

Comment:
I pointed out where the cost value comes from because if this error is raised from `bcrypt::hash()`, it was actually a programmer error, and if you have a hardcoded cost you can just handle that branch with `unreachable!()`. But during verification it means the hash was malformed which is something you have to deal with always.